### ISSUE TITLE
`UFuncTypeError` test failures for Chandrasekhar_G with numpy-dev and xarray-dev

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#1224]
We've been getting some recurrent test failures, in particular a `UFuncTypeError` in `test_Chandrasekhar_G.py`, with the error message below.  These are happening only for numpy-dev and xarray-dev.  

```
------------------------------------------------------------------------------------ Hypothesis ------------------------------------------------------------------------------------
Falsifying explicit example: test_Chandrasekhar_with_hypothesis(
    x=2.220446049250313e-16,
)
Traceback (most recent call last):
  File "/home/runner/work/PlasmaPy/PlasmaPy/plasmapy/formulary/tests/test_Chandrasekhar_G.py", line 22, in test_Chandrasekhar_with_hypothesis
    result = Chandrasekhar_G(x)
numpy.core._exceptions._UFuncNoLoopError: ufunc 'Chandrasekhar_G' did not contain a loop with signature matching types <class 'numpy.dtype[float64]'> -> None

Falsifying explicit example: test_Chandrasekhar_with_hypothesis(
    x=1.7976931348623157e+308,
)
Traceback (most recent call last):
  File "/home/runner/work/PlasmaPy/PlasmaPy/plasmapy/formulary/tests/test_Chandrasekhar_G.py", line 22, in test_Chandrasekhar_with_hypothesis
    result = Chandrasekhar_G(x)
numpy.core._exceptions._UFuncNoLoopError: ufunc 'Chandrasekhar_G' did not contain a loop with signature matching types <class 'numpy.dtype[float64]'> -> None

Falsifying explicit example: test_Chandrasekhar_with_hypothesis(
    x=-1.7976931348623157e+308,
)
Traceback (most recent call last):
  File "/home/runner/work/PlasmaPy/PlasmaPy/plasmapy/formulary/tests/test_Chandrasekhar_G.py", line 22, in test_Chandrasekhar_with_hypothesis
    result = Chandrasekhar_G(x)
numpy.core._exceptions._UFuncNoLoopError: ufunc 'Chandrasekhar_G' did not contain a loop with signature matching types <class 'numpy.dtype[float64]'> -> None

Falsifying explicit example: test_Chandrasekhar_with_hypothesis(
    x=0,
)
Traceback (most recent call last):
  File "/home/runner/work/PlasmaPy/PlasmaPy/plasmapy/formulary/tests/test_Chandrasekhar_G.py", line 22, in test_Chandrasekhar_with_hypothesis
    result = Chandrasekhar_G(x)
numpy.core._exceptions._UFuncNoLoopError: ufunc 'Chandrasekhar_G' did not contain a loop with signature matching types <class 'numpy.dtype[int64]'> -> None

Falsifying explicit example: test_Chandrasekhar_with_hypothesis(
    x=3.761264e-20,
)
Traceback (most recent call last):
  File "/home/runner/work/PlasmaPy/PlasmaPy/plasmapy/formulary/tests/test_Chandrasekhar_G.py", line 22, in test_Chandrasekhar_with_hypothesis
    result = Chandrasekhar_G(x)
numpy.core._exceptions._UFuncNoLoopError: ufunc 'Chandrasekhar_G' did not contain a loop with signature matching types <class 'numpy.dtype[float64]'> -> None
__________________________________________________________________________ test_Chandrasekhar_regression ___________________________________________________________________________
[gw0] linux -- Python 3.9.6 /home/runner/work/PlasmaPy/PlasmaPy/.tox/py39-numpydev/bin/python

num_regression = <pytest_regressions.num_regression.NumericRegressionFixture object at 0x7f285660e9a0>

    def test_Chandrasekhar_regression(num_regression):
        x = np.linspace(-5, 5, 100)
>       y = Chandrasekhar_G(x)
E       numpy.core._exceptions._UFuncNoLoopError: ufunc 'Chandrasekhar_G' did not contain a loop with signature matching types <class 'numpy.dtype[float64]'> -> None

plasmapy/formulary/tests/test_Chandrasekhar_G.py:32: UFuncTypeError
__________________________________________________________________________ test_limiting_behavior_small_x __________________________________________________________________________
[gw0] linux -- Python 3.9.6 /home/runner/work/PlasmaPy/PlasmaPy/.tox/py39-numpydev/bin/python

    def test_limiting_behavior_small_x():
        x = np.logspace(-7, -4, 100)
>       y = Chandrasekhar_G(x)
E       numpy.core._exceptions._UFuncNoLoopError: ufunc 'Chandrasekhar_G' did not contain a loop with signature matching types <class 'numpy.dtype[float64]'> -> None

plasmapy/formulary/tests/test_Chandrasekhar_G.py:38: UFuncTypeError
__________________________________________________________________________ test_limiting_behavior_high_x ___________________________________________________________________________
[gw0] linux -- Python 3.9.6 /home/runner/work/PlasmaPy/PlasmaPy/.tox/py39-numpydev/bin/python

    def test_limiting_behavior_high_x():
        x = np.logspace(6, 90, 100)
>       y = Chandrasekhar_G(x)
E       numpy.core._exceptions._UFuncNoLoopError: ufunc 'Chandrasekhar_G' did not contain a loop with signature matching types <class 'numpy.dtype[float64]'> -> None

plasmapy/formulary/tests/test_Chandrasekhar_G.py:49: UFuncTypeError
```

**Comment by rocco8773** [comment]
Note, these do not just occur for the dev tests.  These failures are related to the `hypothesis` tests and seem to be sporadic.  If you manually re-run the testing CI, then they'll likely pass.  If you re-run them again, they'll likely fail.

This is likely something for @StanczakDominik to look over once his hectic schedule loosens up.

**Comment by namurphy** [comment]
I believe that these are different than the earlier sporadic test failures, which were fixed in #1125.  I've noticed this particular failure only for numpy-dev and xarray-dev, at least so far.  There have been some [PRs related to ufuncs to NumPy](https://github.com/numpy/numpy/pulls?q=is%3Apr+is%3Aclosed+ufunc) merged recently, so my guess is that these are related to that.  But, I haven't looked at this in depth yet so I'm not really sure.
